griptape/configs/drivers/base_drivers_config.py

```diff
@@ -48,6 +48,22 @@ class BaseDriversConfig(ABC, SerializableMixin):
         default=None, kw_only=True, metadata={"serializable": True}, alias="audio_transcription_driver"
     )
 
+    _old_drivers_config: BaseDriversConfig = field(default=None)
+
+    def __enter__(self) -> BaseDriversConfig:
+        from griptape.configs import Defaults
+
+        self._old_drivers_config = Defaults.drivers_config
+
+        Defaults.drivers_config = self
+
+        return self
+
+    def __exit__(self, type, value, traceback) -> None:  # noqa: ANN001, A002
+        from griptape.configs import Defaults
+
+        Defaults.drivers_config = self._old_drivers_config
+
```

Comment:
probably should add `self._old_drivers_config = None`

Reply:
Updated